src/Castle.Facilities.AspNetCore/WindsorRegistrationExtensions.cs

```diff
@@ -0,0 +1,184 @@
+	using Castle.Facilities.AspNetCore.Activators;
+	using Castle.Facilities.AspNetCore.Contributors;
+	using Castle.MicroKernel.Lifestyle;
+	using Castle.MicroKernel.Registration;
+	using Castle.MicroKernel.Resolvers.SpecializedResolvers;
+	using Castle.MicroKernel;
+	using Castle.Windsor;
+
+	using Microsoft.AspNetCore.Builder;
+	using Microsoft.AspNetCore.Http;
+	using Microsoft.AspNetCore.Mvc;
+	using Microsoft.AspNetCore.Razor.TagHelpers;
+	using Microsoft.Extensions.DependencyInjection;
+
+	public static class WindsorRegistrationExtensions
+	{
+		/// <summary>
+		/// This exposes Windsor components to the IServiceCollection using the `CrossWired` registration extension. 
+		/// </summary>
+		/// <param name="services">ASP.NET Core service collection from Microsoft.Extensions.DependencyInjection</param>
+		/// <param name="container">Windsor container which IServiceCollection resolves from using factory methods</param>
+		public static void CrossWiresInto(this IWindsorContainer container, IServiceCollection services)
+		{
+			InstallCrossWiringComponentModelContributor(container.Kernel, services);
+		}
+
+		/// <summary>
+		/// Sets up framework level activators for Controllers, TagHelpers and ViewComponents and installs additional framework integration. This method should be called after everything has been registered.
+		/// </summary>
+		/// <param name="services">ASP.NET Core service collection from Microsoft.Extensions.DependencyInjection</param>
+		/// <param name="container">Windsor container which activators call resolve against</param>
+		/// <param name="configure">Configuration options for controlling registration and lifestyles of controllers, tagHelpers and viewComponents</param>
+		/// <param name="serviceProviderFactory">Optional factory for creating a custom service provider</param>
+		public static IServiceProvider AddWindsor(this IServiceCollection services, IWindsorContainer container, Action<WindsorRegistrationOptions> configure = null, Func<IServiceProvider> serviceProviderFactory = null)
+		{
+			InstallWindsorIntegration(services, container);
+			InstallFrameworkIntegration(services, container);
+
+			var options = new WindsorRegistrationOptions();
+			configure?.Invoke(options);
+
+			AddApplicationComponentsToWindsor(container, options);
+
+			return InitialiseFrameworkServiceProvider(services, serviceProviderFactory, container);
+		}
+
+		/// <summary>
+		/// For making services registered in Windsor visible to the IServiceCollection using 'late bound' factory methods. This makes things like the @Inject directive in Razor work.
+		/// </summary>
+		/// <param name="registration">The component registration that gets copied across to the IServiceCollection</param>
+		public static ComponentRegistration CrossWired(this ComponentRegistration registration)
+		{
+			registration.Activator<CrossWiredComponentActivator>();
+			registration.Attribute("aspnet-core-cross-wired").Eq("yes");
```

Comment:
Just in case you didn't already know, there is an `AddAttributeDescriptor` method that does the same thing. You could also use `bool.TrueString` as your constant if you like.

```
componentRegistration
    .AddAttributeDescriptor(TypedFactoryFacility.IsFactoryKey, bool.TrueString);
```